src/java/org/apache/cassandra/repair/RepairJob.java

```diff
@@ -215,6 +225,10 @@ public void onFailure(Throwable t)
             for (int i = 0; i < trees.size(); i++)
             {
                 InetAddressAndPort address = trees.get(i).endpoint;
+
+                if (isTransient(address))
```

Comment:
So what policy is this implementing for optimizedSyncing? It's still going to look for differences, but only differences that involve a full replica since the differences are still encoded in reduced differences? Maybe add a comment saying the effect here is that we won't stream to transient replicas, but if we come across a full replica we are still comparing it to transient replicas.